components/TextFields/src/MDCTextInput.h

```diff
@@ -0,0 +1,143 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+/**
+ The Material Design guideslines have many suggestions for handling text input. The inputs that
+ conform to this protocol have all the API necessary to achieve those suggestions.
+
+ They are, however, dumb; they do not handle error state nor validation.
+
+ - For handling error states and other Material behaviors use an MDCTextInputController on your
+ inputs.
+ - For validation, there are many 3rd party libraries you can use like:
+   - https://github.com/3lvis/Validation
+   - https://github.com/adamwaite/Validator
+ */
+
+@protocol MDCTextInputPositioningDelegate;
+
+/** Common API for Material Design compliant text inputs. */
+@protocol MDCTextInput <NSObject>
```

Comment:
Well, I did it this way to match the [Swift API guidelines](https://swift.org/documentation/api-design-guidelines/#strive-for-fluent-usage). They have set a new standard where protocols and classes are intentionally often named the same. This protocol doesn't describe just functionality like a delegate protocol. It describes a class with the following setters and getters (and some methods.) 
>Protocols that describe what something is should read as nouns (e.g. Collection).